### PR TITLE
cli: default the log directory to the "logs" subdirectory of the first store

### LINKED ISSUE #4492: cli: default log directory to the first store directory
Rather than default the log directory to `os.TempDir` (i.e. `${TMPDIR}`) we could default it to a subdirectory of the first store directory (i.e. `./cockroach-data/logs`). This would make the logs slightly easier to access. Any concerns with doing this? Is `${TMPDIR}` preferable for some reason? One downside to `${TMPDIR}` is that logs tend to accumulate there without bound.

If there are no store directories we could fall back to using `${TMPDIR}` (e.g. for `./cockroach sql`).

### PR DESCRIPTION
**Pull request by petermattis**
Tweak the info that is output at startup.

Fixes #4492.

```
~ ./cockroach start --insecure
build:     alpha.v1-226-gefd2c41 @ 2016/02/19 02:43:06 (go1.6)
admin:     http://Peters-MacBook-Pro-2.local:26257
sql:       postgresql://root@Peters-MacBook-Pro-2.local:15432?sslmode=disable
logs:      cockroach-data/logs
store[0]:  cockroach-data
```

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4505)

### CODE REVIEW

**Comment by nvanbenschoten** [comment]
Reviewed 1 of 3 files at r1.
Review status: 1 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[cli/start.go, line 115 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4505#-KAvf01htqYqaexLAhr4:-KAvf01htqYqaexLAhr5:998920311) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a74d958fa3a378355a367812fcc39a24b30728f/cli/start.go#L115)):*
Why is this logic separate from `cli/flags.go:203`? Couldn't you do the correct defaulting from the above comment in the other file? Is the reason because we only want this behavior for the `start` command?

---

*[cli/start.go, line 164 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4505#-KAvfsGqn0B63s3p71KG:-KAvfsGqn0B63s3p71KH:-188920334) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a74d958fa3a378355a367812fcc39a24b30728f/cli/start.go#L164)):*
Why not panic on an error instead of throwing it away?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4505)*

File: cli/start.go [hunk 1]
```diff
@@ -99,6 +101,33 @@ uninitialized, specify the --join flag to point to any healthy node
...
 // cluster, if this is its first time connecting.
 func runStart(_ *cobra.Command, _ []string) error {
+	storeSpecs, err := cliContext.GetStoreSpecs()
+	if err != nil {
+		return err
+	}
+	// Default the log directory to the the "logs" subdirectory of the first
+	// non-memory store.
+	//
+	// TODO(pmattis): This isn't quite correct. The user might specify
```
Comment (tschottdorf) [inline cli/start.go]: File this as easy+helpwanted.

**Comment by petermattis** [comment]
Review status: 1 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[cli/start.go, line 111 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4505#-KArPx94DBSsGuLNPAl1-r1-111:-KAvh8JmbBUQJ8AR1fIb:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a74d958fa3a378355a367812fcc39a24b30728f/cli/start.go#L111)):*
Done.

---

*[cli/start.go, line 115 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4505#-KAvf01htqYqaexLAhr4:-KAvgLLFWnSEkz4Jyfaj:-1569983078) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a74d958fa3a378355a367812fcc39a24b30728f/cli/start.go#L115)):*
Yes, we only wants this behavior for `start`. Or, at least, that is my intention.

---

*[cli/start.go, line 164 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4505#-KAvfsGqn0B63s3p71KG:-KAvgSenXWo9by2CcZZO:-921142542) ([raw file](https://github.com/cockroachdb/cockroach/blob/2a74d958fa3a378355a367812fcc39a24b30728f/cli/start.go#L164)):*
I don't think any errors are possible. Regardless, seems fine to return the error if it occurs. Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4505)*

File: cli/start.go [hunk 2]
```diff
@@ -124,10 +153,15 @@ func runStart(_ *cobra.Command, _ []string) error {
 		return fmt.Errorf("cockroach server exited with error: %s", err)
 	}
 
-	fmt.Printf("admin:  %s\n", cliContext.AdminURL())
-	fmt.Printf("sql:    %s\n", cliContext.PGURL(connUser))
-	fmt.Printf("logs:   %s\n", flag.Lookup("log-dir").Value)
-	fmt.Printf("stores: %s\n", cliContext.Stores)
+	tw := tabwriter.NewWriter(os.Stdout, 2, 1, 2, ' ', 0)
```
Comment (tschottdorf) [inline cli/start.go]: :+1:

**Comment by nvanbenschoten** [comment]
:lgtm: 

---

Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4505#-:-KAvmen7XkFWNWUmDlM-:980077423)*

**Comment by jseldess** [comment]
Covered in **Logging Flags** section of [Cockroach Commands](https://www.cockroachlabs.com/docs/cockroach-commands.html).

**Pull request merged**